src/util.ts

```diff
@@ -39,77 +39,73 @@ export function nop() {
+}
+/**
+ *
+ * Recursively iterate through the dependency chain until we reach the end of
+ * the dependency chain or encounter a circular reference
+ * @param filePath Filepath of file currently being read
+ * @param customReadFilep the file reading function being used
+ * @param readFiles an array of the previously read files so we can check for
+ * circular references returns a ConfigFile object containing the data from all
+ * the dependencies
+ */
+async function getBase(
+    filePath: string, customReadFilep: ReadFileP, readFiles: string[],
+    currentDir: string): Promise<ConfigFile> {
   customReadFilep = customReadFilep || readFilep;
-  const json = await customReadFilep(tsconfigPath, 'utf8');
-  let contents = JSON.parse(json);
 
-  const readArr = ['tsconfig.json'];
-  if (contents['extends']) {
-    const extension =
-        await getExtension(contents['extends'], customReadFilep, readArr);
-    contents = combineTSConfig(extension, contents);
-  }
-  return Promise.resolve(contents);
-}
+  filePath = path.resolve(currentDir, filePath);
 
-async function getExtension(
-    filePath: string, customReadFilep: ReadFileP,
-    readFiles: string[]): Promise<ConfigFile> {
-  customReadFilep = customReadFilep || readFilep;
-  if (!filePath) {
-    throw new Error('Undefined passed in');
-  }
   if (readFiles.indexOf(filePath) !== -1) {
-    // did I throw the error correctly?
-    throw new Error('Circular Reference Detected');
+    throw new Error(
+        'Circular reference in ' + readFiles[readFiles.indexOf(filePath)]);
   }
   readFiles.push(filePath);
+
   const json = await customReadFilep(filePath, 'utf8');
   let contents = JSON.parse(json);
 
-  if (contents['extends']) {
-    const nextFile =
-        await getExtension(contents['extends'], customReadFilep, readFiles);
+  if (contents.extends) {
+    const nextFile = await getBase(contents.extends, customReadFilep, readFiles,
+                                                   path.dirname(filePath));
     contents = combineTSConfig(nextFile, contents);
   }
-  // console.log(contents);
-  return Promise.resolve(contents);
+
+  return contents;
 }
 
-// the inherited config file overwrites the base config file's "files",
-// "include", and "exclude" fields and combines compiler options
+/**
+ * takes in 2 config files
+ * @param base is loaded first
+ * @param inherited is then loaded and overwrites base
+ */
 function combineTSConfig(base: ConfigFile, inherited: ConfigFile): ConfigFile {
-  // const TSProperties = ["compilerOptions", "files", "include", "exclude"];
-  const result = {
-    'compilerOptions': {},
-    'files': [],
-    'include': [],
-    'exclude': [],
-    'extends': {}
-  };
+  const result = {'compilerOptions': {}, 'extends': {}};
 
   Object.assign(result, base, inherited);
   Object.assign(
       result.compilerOptions, base.compilerOptions, inherited.compilerOptions);
   delete result.extends;
   return result;
-}
\ No newline at end of file
+}
```

Comment:
Nit: Add a line between a class/interface definition and the docs for the next definition. (I think this is also true of `getBase`)